webapp/src/app/newuser/newuser.component.ts

```diff
@@ -1,7 +1,102 @@
 import { Component, OnInit } from "@angular/core";
+import { FormControl, FormGroup, Validators } from "@angular/forms";
+import { MatSnackBar } from "@angular/material/snack-bar";
+import { MatDialog } from "@angular/material/dialog";
+import PersonhoodRPC from "@dedis/cothority/personhood/personhood-rpc";
+import { ByzCoinService } from "src/app/byz-coin.service";
+import Log from "@dedis/cothority/log";
+import { showDialogInfo, showTransactions, TProgress } from "src/lib/Ui";
+import { ServerIdentity } from "@dedis/cothority/network";
 
 @Component({
-  selector: "app-newuser",
-  templateUrl: "./newuser.component.html",
+    selector: "app-newuser",
+    templateUrl: "./newuser.component.html",
 })
-export class NewuserComponent {}
+export class NewuserComponent implements OnInit {
+    signupForm: FormGroup;
+    recoveryForm: FormGroup;
+
+    constructor(private snack: MatSnackBar,
+                private dialog: MatDialog,
+                public bcs: ByzCoinService) {
+    }
+
+    async ngOnInit() {
+        for (const node of this.bcs.config.roster.list) {
+            Log.info("node", node.description, "id", node.id);
+        }
+        this.signupForm = new FormGroup({
+            alias: new FormControl(),
+            email: new FormControl("", Validators.email),
+        });
+        this.recoveryForm = new FormGroup({
+            email: new FormControl("", Validators.email),
+        });
+    }
+
+    getPHRPC(): PersonhoodRPC {
+        if (this.bcs.config.signupNode === "") {
```

Comment:
the field is `undefined` if it is not existing in the config, not the empty string

```suggestion
        if (this.bcs.config.signupNode === undefined) {
```